controllers/keda/triggerauthentication_controller.go

```diff
@@ -116,3 +121,80 @@ func (r *TriggerAuthenticationReconciler) UpdatePromMetricsOnDelete(namespacedNa
 
 	delete(triggerAuthPromMetricsMap, namespacedName)
 }
+
+func GetTriggerAuth(ctx context.Context, client client.Client, triggerAuthRef *kedav1alpha1.ScaledObjectAuthRef, namespace string) (client.Object, error) {
+	if triggerAuthRef == nil {
+		return nil, fmt.Errorf("triggerAuthRef is nil")
+	}
+
+	if triggerAuthRef.Kind == "" || triggerAuthRef.Kind == "TriggerAuthentication" {
+		triggerAuth := &kedav1alpha1.TriggerAuthentication{}
+		err := client.Get(ctx, types.NamespacedName{Name: triggerAuthRef.Name, Namespace: namespace}, triggerAuth)
+		if err != nil {
+			return nil, err
+		}
+		return triggerAuth, nil
+	} else if triggerAuthRef.Kind == "ClusterTriggerAuthentication" {
+		clusterTriggerAuth := &kedav1alpha1.ClusterTriggerAuthentication{}
+		err := client.Get(ctx, types.NamespacedName{Name: triggerAuthRef.Name, Namespace: namespace}, clusterTriggerAuth)
+		if err != nil {
+			return nil, err
+		}
+		return clusterTriggerAuth, nil
+	}
+	return nil, fmt.Errorf("unknown trigger auth kind %s", triggerAuthRef.Kind)
+}
+
+func GetTriggerAuthStatus(triggerAuth client.Object) (*kedav1alpha1.TriggerAuthenticationStatus, error) {
```

Comment:
both these funcitons are called only once in the `UpdateTriggerAuthenticationStatusHandler`, I don't think it is necessary to create a new function for each (even public ones). I think you can simplify and merge the code to query both TriggerAuth and TriggerAuth Status in the single function or command `UpdateTriggerAuthenticationStatusHandler`